src/components/auth/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import InputMessage from "../InputMessage";
+import { TextInput } from "../TextInput";
+import { Button } from "../ui/button";
+
+interface SignUpFormData {
+  email: string;
+  password: string;
+}
+
+const emailSchema = z.string().email({ message: "Invalid email" });
+const passwordSchema = z
+  .string()
+  .min(8, { message: "Password must be at least 8 characters" })
+  .regex(/\d/, { message: "Password must contain a number" })
+  .regex(/[a-z]/, { message: "Password must contain a lowercase letter" })
+  .regex(/[A-Z]/, { message: "Password must contain an uppercase letter" });
+
+export const EmailSignUp = () => {
+  const [success, setSuccess] = useState<boolean>(false);
+  const [buttonDisabled, setButtonDisabled] = useState<boolean>(false);
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<SignUpFormData>();
```

Comment:
use zodResolver for validation, and provide the error messages described in the schema as the textinput error messages.

something similar was done in the SemesterEditDialog component so you could check that out